src/Compilers/CSharp/Test/Semantic/SourceGeneration/GeneratorDriverTests.cs

```diff
@@ -902,6 +902,109 @@ class C
                );
             Assert.Same(oldDriver, driver);
         }
+
+        [Fact]
+        public void DeveloperMode_Enables_Different_Exception_Warning_During_Excecution()
+        {
+            // non developer mode:
+            var results = runGenerator(asDeveloper: false);
+
+            results.Diagnostics.Verify(
+                 // warning CS8785: Generator 'CallbackGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result.
+                 Diagnostic(ErrorCode.WRN_GeneratorFailedDuringGeneration).WithArguments("CallbackGenerator", "InvalidOperationException", "I'm sorry dave").WithLocation(1, 1)
```

Comment:
>, "InvalidOperationException", "I'm sorry dave" [](start = 108, length = 47)

I'm not sure why non-developer version would have these as arguments.

Reply:
Because we still show some exception info in the description of the non-dev one. Just not stack traces etc.

Reply:
Is the message wrong then?

---
In reply to: [469431935](https://github.com/dotnet/roslyn/pull/46714#discussion_r469431935) [](ancestors = 469431935)